src/tests/suite/daos_dtx.c

```diff
@@ -306,21 +308,10 @@ dtx_fetch_committable(void **state, bool punch)
 	daos_fail_loc_set(DAOS_OBJ_SPECIAL_SHARD | DAOS_FAIL_ALWAYS);
 
 	rc = dtx_check_replicas_v2(dkey, akey, "fetch_committable_1",
-				   update_buf2, punch ? 0 : dts_dtx_iosize / 2,
-				   &req);
+				   punch ? zero_buf : update_buf2,
+				   dts_dtx_iosize / 2, &req);
 	/* At least leader will return the latest data. */
 	assert_true(rc >= 1);
-	if (rc > 1) {
-		assert_true(!retried);
-
-		/* Related DTX has been committed asychronously. If we modify
-		 * again, according to current DTX batched commit policy, the
-		 * new DTX will NOT be committed so quickly again.
-		 */
-		retried = true;
-		dts_buf_render(update_buf2, dts_dtx_iosize / 2);
-		goto again;
-	}
```

Comment:
Why remove this? retry with leader when read from non-leader is not related with punch logic.

Reply:
The test would fail consistently with this code and I was unable to figure out what it was trying to do.

Reply:
The original logic is that:
The dtx_check_replicas_v2() return value is the number for replicas that returned expected (latest update) data. The object has 3 replicas, so @rc can be 0/1/2/3. The leader will always return the latest data (in spite of related DTX is committable or committed), so @rc will be 1/2/3. "(rc > 1)" means some non-leader also returned latest data, so means that related DTX has been committed or in-committing. According to our current batched commit policy, if we make another modification, the new DTX will not be committed immediately. Then the next check dtx_check_replicas_v2 via "goto again" should return 1.

Reply:
It would appear the logic has changed in some manner.  If I add it back in, the assertion triggers.